apps/VendingMachine/contracts/VendingMachine.ral

```diff
@@ -0,0 +1,174 @@
+import "std/nft_collection_interface"
+import "std/nft_collection_with_royalty_interface"
+
+Interface IRandomNumberGenerator{
+    @using(updateFields = true,checkExternalCaller = false)
+    pub fn getRandomNumber(userHash_: ByteVec) -> (U256)
+}
+
+Contract VendingMachine(
+    foodsContractId: ByteVec,
+    collectionOwner: Address,
+    collectionUri: ByteVec,
+    mut nftBaseUri: ByteVec,
+    mut totalSupply: U256, 
+    mut mintIsPaused: Bool // true
+) implements INFTCollectionWithRoyalty {
+
+    event NftMinted(minter: Address, id: ByteVec, index: U256)
+
+    // const CollectionUri = b`Vending Machine`
+    const MaxSupply = 1000
+    const MintPrice = 1 alph const // 
+    RoyaltyRate = 500 // 5 %  == 500 basis point // 
+
+    enum ErrorCodes {
+        MaxSupplyReached = 0
+        OnlyCollectionOwner = 3
+        MintIsPaused = 5
+        NFTNotPartOfCollection = 7
+        NotMintedYet = 8
+    }
+
+    @using(checkExternalCaller = false)
+    pub fn getRandom() -> U256{
+        return IRandomNumberGenerator(addressToContractId!(@xeCgiV9x2sFysJiubJZjviUoaCEkVofeBJeUh4B1EF5R)).getRandomNumber(toByteVec!(callerAddress!()))
```

Comment:
is this address for mainnet? probably want to pass this in as a parameter I think?